src/Adapters/IndexManagerAdapter.php

```diff
@@ -107,6 +107,17 @@ public function dropIfExists(string $indexName): IndexManagerInterface
         return $this;
     }
 
+    public function dropAll(): IndexManagerInterface
```

Comment:
Unfortunately I don't use prefixes in my environment, so using `*` would drop my Kibana indexes as well.

> The only problem, that `drop` doesn't process patterns correctly (because of prefixing)

Shouldn't we expand this use-case if it cannot process patterns correctly? A `dropAll()` method would resolve this issue.

Also, the `fresh` command is explicit as to what it does. When you run `migrate:fresh`, you expect the whole database reset to its original state. I'd expect `elastic:migrate:fresh` to perform the same task.

Thoughts?

Reply:
> Also, the fresh command is explicit as to what it does. When you run migrate:fresh, you expect the whole database reset to its original state.

I agree, but even you want to exclude something with this pattern `-*.*` 🙂 I'm afraid to hard code something like this `*,-*.*`, cause it works for you, but maybe won't work for others. It's better to have a generic solution.

> Unfortunately I don't use prefixes in my environment, so using * would drop my Kibana indexes as well.

Why can't you configure a prefix for your test environment? You can do it in `phpunit.xml.dist` by defining an env variable for example.

I still prefer to drop indices via `adapter->drop('*')`. Users expect that this package creates and drops only prefixed indices, it's supported by all methods.

Reply:
> Why can't you configure a prefix for your test environment? You can do it in `phpunit.xml.dist` by defining an env variable for example.

Why do I need to have a prefix? Shouldn't this be an optional feature? The default configured prefix is an empty string, so I should be able to use this package and all of its features without having one, right?

> I still prefer to drop indices via `adapter->drop('*')`. Users expect that this package creates and drops only prefixed indices, it's supported by all methods.

Ok -- I've just pushed an update for this.

Reply:
> Why do I need to have a prefix? Shouldn't this be an optional feature? The default configured prefix is an empty string, so I should be able to use this package and all of its features without having one, right?

Correct, but this command now does what is expected - it drops all indices and then runs migrations. Everything works with the default prefix, but if you have a problem - you need to differentiate indices created by this library from system indices. This problem can be solved by using a prefix, this is a valid use case from my point of view, don't you agree?

Reply:
Yea you're right -- that makes sense to me, thanks for taking the time to walk me through the explanation 👍

Reply:
@stevebauman sorry for taking so long to review/merge this ticket, I'll create a new release and include the new command in a week or so, when I'm back from vacation.